source/server/http_route_manager_impl.cc

```diff
@@ -0,0 +1,34 @@
+#include "server/http_route_manager_impl.h"
+
+namespace Envoy {
+namespace Server {
+
+std::vector<Router::RouteConfigProviderSharedPtr> HttpRouteManagerImpl::routeConfigProviders() {
+  std::vector<Router::RouteConfigProviderSharedPtr> ret;
+  ret.reserve(route_config_providers_.size());
+  for (const auto& element : route_config_providers_) {
+    ret.push_back(element.second.lock());
```

Comment:
The whole implementation will need to change a bit so let's touch on this later.